### ISSUE TITLE
ComplexMethod ignore mapper or allow for not counting kotlin elvis operator 

### ISSUE THREAD

**Issue by DcortezMeleth** [detekt/detekt#3502]
There is issue [3103](https://github.com/detekt/detekt/issues/3103) which tackles similar problem but it could be solved in a way this case can't.

## Expected Behavior of the rule
Having mapper like
```
fun map(employee: ExternalApiEmployee): Employee = Employee(
                    employee.id ?: "",
                    employee.job ?: "",
                    employee.getName()?.let {
                        (it.knownAs ?: it.firstName ?: "") + " " + (it.lastName ?: "")
                    } ?: "",
                    employee.dateOfBirth ?: LocalDate.of(1990,1,1) 
                    employee.active ?: false)
```
I would like to have a way to make ComplexMethod rule to skip it. I'm aware that this example has lower MCC number than default threshold but it is just an idea, as we have much bigger mapper and our project but I don't think it is necessary to show it all here.

Two ways I could think of is either allowing some regex for class names which can be excluded. Or maybe some flag allowing for not counting elvises to MCC, but it may create some false-negatives.
Also it may best to just suppress this rule in those cases and don't provide direct way for this, as this may not be most common use case.

## Context
Some of our mappers uses some nullable external APIs and require a lot of default values in mapper.
We then map those values in mappers with elvis operators. Because of this we have to suppress ComplexMethod rule on this classes. We don't feel as this elvises make our method really complex in this context also there is no other way to achieve it

**Comment by schalkms** [comment]
> as this may not be most common use case.

As you said, this is a perfect example for implementing a custom rule. You could copy the existing rule logic and adapt it to your needs.

In my opinion, the mapper code is complex according to the MCC.

**Comment by schalkms** [comment]
> as this may not be most common use case.

As you said, this is a perfect example for implementing a custom rule. You could copy the existing rule logic and adapt it to your needs.

The mapper code is complex according to the MCC.
